### ISSUE TITLE
Xtb error with silicious compounds

### ISSUE THREAD

**Issue by chemistza** [grimme-lab/xtb#303]
When trying to do single point calculations on some silicious compounds (obtained by cutting various sized clusters from MFI zeolite and terminating dangling oxygen atoms with H) xtb (version 6.3.2) crashes with a segmentation fault. It also states that the molecule is fragmented (which it isn't). For smaller clusters it works, but as soon as you go beyond a certain size it fails. Below is sample output for a failed case (.xyz file attached as a zip file).

```
 * started run on 2020/07/13 at 11:10:03.831     

           -------------------------------------------------
          |                Calculation Setup                |
           -------------------------------------------------

          program call               : xtb SiOH.xyz xtb
          coordinate file            : SiOH.xyz
          omp threads                :                     6
          number of atoms            :                    96
          number of electrons        :                   392
          charge                     :                     0
          spin                       :                   0.0
          first test random number   :      0.79899123955731

   ID    Z sym.   atoms
    1    8 O      1, 3, 4, 6-10, 12, 14, 16, 18-20, 28, 29, 31, 33, 35, 37,
                  39, 41, 43, 46, 48, 50, 52, 55, 57, 58, 60, 62, 64, 66, 69,
                  71, 73, 75, 76, 79-82, 86-88, 92, 94, 95
    2    1 H      2, 5, 11, 13, 15, 17, 21, 30, 32, 34, 38, 40, 42, 44, 47,
                  49, 51, 53, 59, 61, 63, 65, 67, 72, 74, 77, 83, 89, 93, 96
    3   14 Si     22-27, 36, 45, 54, 56, 68, 70, 78, 84, 85, 90, 91

molecular fragmentation (1/2 indicates fragments):
111111111111111111111111111111111111111111111111111111111111111111111111
111111111221111111111122
# atoms in fragment 1/2:    92     4
 fragment masses (1/2) :     1257.65       34.01
CMA distance (Bohr)    :  11.883
constraining FC (au)   :  0.0500

           -------------------------------------------------
          |                 G F N 2 - x T B                 |
           -------------------------------------------------

        Reference                      10.1021/acs.jctc.8b01176
      * Hamiltonian:
        H0-scaling (s, p, d)           1.850000    2.230000    2.230000
        zeta-weighting                 0.500000
      * Dispersion:
        s8                             2.700000
        a1                             0.520000
        a2                             5.000000
        s9                             5.000000
      * Repulsion:
        kExp                           1.500000    1.000000
        rExp                           1.000000
      * Coulomb:
        alpha                          2.000000
        third order                    shell-resolved
        anisotropic                    true
        a3                             3.000000
        a5                             4.000000
        cn-shift                       1.200000
        cn-exp                         4.000000
        max-rad                        5.000000


          ...................................................
          :                      SETUP                      :
          :.................................................:
          :  # basis functions                 396          :
          :  # atomic orbitals                 379          :
          :  # shells                          179          :
          :  # electrons                       392          :
          :  max. iterations                   250          :
          :  Hamiltonian                  GFN2-xTB          :
          :  restarted?                      false          :
          :  GBSA solvation                  false          :
          :  PC potential                    false          :
          :  electronic temp.          300.0000000     K    :
          :  accuracy                    1.0000000          :
          :  -> integral cutoff          0.2500000E+02      :
          :  -> integral neglect         0.1000000E-07      :
          :  -> SCF convergence          0.1000000E-05 Eh   :
          :  -> wf. convergence          0.1000000E-03 e    :
          :  Broyden damping             0.4000000          :
          ...................................................
Segmentation fault (core dumped)
```

[SiOH.zip](https://github.com/grimme-lab/xtb/files/4911472/SiOH.zip)

**Comment by awvwgk** [comment]
I can reproduce the failure with system defaults, but it goes away when increasing the `OMP_STACKSIZE`:

```
OMP_STACKSIZE=1G xtb SiOH.xyz
```

The above runs fine on four threads on my notebook. Please check the documentation about settings for the parallelisation:
https://xtb-docs.readthedocs.io/en/latest/setup.html#parallelisation

**Comment by chemistza** [comment]
That is weird, on my system I have:

```
export OMP_STACKSIZE=2G
export OMP_NUM_THREADS=6,1
export OMP_MAX_ACTIVE_LEVELS=1
export MKL_NUM_THREADS=6
```
And I still get the error. I have also tried larger and smaller values for OMP_STACKSIZE, but the error persists.

**Comment by awvwgk** [comment]
It should run always with only one thread regardless of the `OMP_STACKSIZE`, try to reduce the number of threads and see if the issue persists.

Memory issues with OMP are quite tricky to handle and will, unfortunately, require some experimenting with the OMP environment variables.

**Comment by chemistza** [comment]
It still crashes with the following settings:
```
export OMP_STACKSIZE=4G
export OMP_NUM_THREADS=1,1
export OMP_MAX_ACTIVE_LEVELS=1
export MKL_NUM_THREADS=1
```

However, if I use the pre-compiled 6.3.2 binaries it works fine on 6 threads and with OMP_STACKSIZE=4G!

Xtb on my system was built with GCC version 10.1.0, meson 0.52.0, and ninja 1.10.0 on Ubuntu 20.04.

Thanks.

**Comment by awvwgk** [comment]
I can reproduce this with GCC 10.1.0 only if I do *not* set `ulimit -s unlimited`. Apparently the stacksize (not the OMP stacksize) is more of an issue with GCC than with Intel.

**Comment by chemistza** [comment]
Great, thanks for your assistance.

**Comment by awvwgk** [comment]
Okay, closing this issue than.

**Comment by sergeyusoltsev** [comment]
Dear colleagues, 
It took me almost infinite time to figure out, but I found a proper solution for the issue on a build stage. Actually, I changed two things (and did not test which exactly of two led me to success):
First, I added
`add_project_arguments('-heap-arrays', language: 'fortran')`
to xtb/meson/meson.build which supposed to force xtb to store arrays in heap instead of stack, didn't worked for me tho, then adding.
`-Wl,-stack_size -Wl,0x40000000 -Wl,-stack_addr -Wl,0xf0000000`
to LD_FLAGS for xtb binary resolved an issue. It increases stack size dedicated to an xtb to 1Gb which overcomes ulimit -s on MacOS. The latter should probably be properly done by appending 
`link_args : '-Wl,-stack_size -Wl,0x40000000 -Wl,-stack_addr -Wl,0xf0000000'`
to xtb/meson.build in the executable() section, but I made a huge mess in my environment and desperately modified build.ninja myself

I am by no means experienced in what I've accomplished here, but this hint may help somebody so I felt urge to write solution here.
